File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {

	/**
	 * 查询所有
	 * @return
	 */
	public List<TbBrand> findAll();
	
	
	/**
	 * 分页查询
	 * @param pageNum  当前页
	 * @param pageSize  每页记录数
	 * @return
	 */
	public PageResult findByPage(int pageNum,int pageSize);
	
	
	/**
	 * 新增方法
	 * @param tbBrands
	 * @return 
	 */
	public void add(TbBrand tbBrands);
	
	
	/**
	 * 根据主键查找指定的品牌
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	
	/**
	 * 修改方法
	 * @param tbBrands
	 * @return 
	 */
	public void update(TbBrand tbBrands);
	
	
	
	/**
	 * 删除方法
	 * @param tbBrands
	 * @return 
	 */
	public void delete(Long[] ids);
	
	/**
	 * 分页查询
	 *  @param tbBrand  参数
	 * @param pageNum  当前页
	 * @param pageSize  每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand  tbBrand,int pageNum,int pageSize);
	
	
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
 //自定义服务
app.service('brandService',function($http){
        	  
        	  //查询全部
     	     this.finadAll = function(){
        		  
     	    	return  $http.get('../brand/findAll.do');
        	    		  
     	     };
     	     
     	     
     	    //分页查询
   	        this.findPage = function(page,rows){
   	    	 
   	    	   return  $http.get('../brand/findByPage.do?pageNum='+page+'&pageSize='+rows);
   	        };
   	      
   	      
   	      
          
          //有条件分页查询
    	  this.search = function(page,rows,searchEntity){
        	  console.log(page);
        	  console.log(rows);
        	  console.log(searchEntity);
    		  return $http.post('../brand/search.do?pageNum='+page+'&pageSize='+rows,searchEntity);
    	  };
          
          //查询指定品牌
          this.findOne=function(id){
    	        $http.get('../brand/findOne.do?id='+id);
          };
          
          
          //新增
          this.add=function(entity){
        	  
        	    return $http.post('../brand/add.do',entity);
          };
          
           //修改
           this.update=function(entity){
        	  
        	    return $http.post('../brand/update.do',entity);
          };
       
          //删除
          this.dele = function(selectIds){
        	  
        	   return $http.get('../brand/delete.do?ids='+selectIds);
          };
    	    
    	      
        	  
    });
          